src/matching.rs

```diff
@@ -209,23 +196,36 @@ fn do_output(badness: HashMap<String, u32>) {
     }
 }
 
-/*
-echo "matching unsafe deps with callgraph nodes"
-python3 "$SIDEROPHILE_PATH/script/find_unsafe_nodes.py" \
-    "$SIDEROPHILE_OUT/unmangled_callgraph.dot" \
-    "$SIDEROPHILE_OUT/unsafe_deps.txt" \
-    > "$SIDEROPHILE_OUT/nodes_to_taint.txt"
-
-echo "tracing the unsafety up the tree"
-python3 "$SIDEROPHILE_PATH/script/trace_unsafety.py" \
-    "$SIDEROPHILE_OUT/unmangled_callgraph.dot" \
-    "$SIDEROPHILE_OUT/nodes_to_taint.txt" \
-    "$CRATENAME" \
-    > "$SIDEROPHILE_OUT/badness.txt"
-*/
-pub fn real_main(args: &TraceArgs) -> CliResult {
-    let callgraph = parse_input_data(&args.input_callgraph, &args.input_unsafe_deps_file);
-    let badness = trace_unsafety(callgraph, &args.crate_name);
+fn do_demangle(input: &PathBuf) -> io::Result<Vec<String>> {
+    let mut in_reader = BufReader::new(File::open(input)?);
+    let mangled_name_regex: Regex = Regex::new(r"_(ZN|R)[\$\._[:alnum:]]*").unwrap();
```

Comment:
we could link to https://en.wikipedia.org/wiki/Name_mangling#Rust or some better resource in a comment

Reply:
added